File: src/main/java/no/nav/syfo/controller/domain/sykepengesoknad/oppsummering/Oppsummering.java
package no.nav.syfo.controller.domain.sykepengesoknad.oppsummering;

import java.io.Serializable;
import java.util.List;

public class Oppsummering implements Serializable {
    public List<Sporsmal> soknad;
    public Sporsmal bekreftetKorrektInformasjon;
    public Tilleggstekst vaerKlarOverAt;

    public Oppsummering withSporsmalsliste(final List<Sporsmal> oppsummering) {
        this.soknad = oppsummering;
        return this;
    }

    public Oppsummering withBekreftetKorrektInformasjon(final Sporsmal bekreftetKorrektInformasjon) {
        this.bekreftetKorrektInformasjon = bekreftetKorrektInformasjon;
        return this;
    }

    public Oppsummering withAnsvarserklaring(final Tilleggstekst ansvarserklaring) {
        this.vaerKlarOverAt = ansvarserklaring;
        return this;
    }
}

File: src/main/kotlin/no/nav/syfo/controller/domain/sykepengesoknad/oppsumering/Oppsummering.kt
package no.nav.syfo.controller.domain.sykepengesoknad.oppsumering

import java.io.Serializable

class Oppsummering : Serializable {
    var soknad: List<Sporsmal>? = null
    var bekreftetKorrektInformasjon: Sporsmal? = null
    var vaerKlarOverAt: Tilleggstekst? = null
    fun withSporsmalsliste(oppsummering: List<Sporsmal>?): Oppsummering {
        soknad = oppsummering
        return this
    }

    fun withBekreftetKorrektInformasjon(bekreftetKorrektInformasjon: Sporsmal?): Oppsummering {
        this.bekreftetKorrektInformasjon = bekreftetKorrektInformasjon
        return this
    }

    fun withAnsvarserklaring(ansvarserklaring: Tilleggstekst?): Oppsummering {
        vaerKlarOverAt = ansvarserklaring
        return this
    }
}
